Sources/TuistCore/Graph/GraphTraverser.swift

```diff
@@ -822,20 +822,15 @@ public class GraphTraverser: GraphTraversing {
         path: AbsolutePath,
         name: String
     ) -> [GraphDependencyReference] {
-        let precompiledStatic = graph.dependencies[.target(name: name, path: path), default: []]
-            .filter { dependency in
-                switch dependency {
-                case let .xcframework(_, _, _, linking: linking):
-                    return linking == .static
-                case .framework, .library, .bundle, .packageProduct, .target, .sdk:
-                    return false
-                }
+        let dependencies = filterDependencies(from: .target(name: name, path: path), test: { dependency in
+            switch dependency {
+            case let .xcframework(_, _, _, linking: linking):
+                return linking == .static
+            case .framework, .library, .bundle, .packageProduct, .target, .sdk:
+                return false
             }
-
-        let precompiledDependencies = precompiledStatic
-            .flatMap { filterDependencies(from: $0) }
-
-        return Set(precompiledStatic + precompiledDependencies)
+        })
```

Comment:
I believe we'll need to obtain all the transitive unlinked xcframeworks as opposed to all, for example:

![example](https://github.com/tuist/tuist/assets/11914919/aba5c2cf-abc3-4d7e-ae11-92e43dbee69d)

This method would only need return xcframeworks `A` and `B`, it shouldn't need `C`

```suggestion
        }, skip: canDependencyLinkStaticProducts)
```